content/en/docs/components/trainer/operator-guides/extension-framework.md

```diff
@@ -0,0 +1,91 @@
+
+### Examples
+
+To illustrate how the framework can be used, consider the following scenarios:
+1. **Custom Validation**: A user wants to add specific validation logic to ensure that certain fields in the `TrainJob` are correctly configured before execution.
+2. **Dynamic Resource Deployment**: A user needs to deploy Kubernetes resources dynamically based on runtime-specific configurations.
+
+## Core Concepts
+
+### Phases
+
+The Kubeflow Trainer Extension Framework follows a structured, step-by-step execution flow. Each phase represents a logical part of the workflow:
+
+- **Startup Phase**: Executes once during the initialization of the `kubeflow-trainer-controller-manager`. This phase sets up necessary internal components.
+- **PreExecution Phase**: Triggered when a `TrainJob` is created or updated. This phase validates and prepares the job for execution.
+- **Build Phase**: Builds and deploys the required Kubernetes resources for training jobs.
+- **PostExecution Phase**: Runs after the job has been built and executed, checking the status and applying relevant conditions to the job.
+
+### APIs and Extension Points
+
+In each phase, there are two types of components:
+1. **Internal APIs**: These APIs are used internally by the framework and cannot be extended or modified by the user.
+2. **Extension Points**: These points are exposed to the user and allow for customization through plugins that can be added to the framework.
+
+<img src="/docs/components/trainer/operator-guides/images/KubeflowTrainerExtensionFramework.drawio.svg"
+  alt="Kubeflow Trainer Extension Framework"
+  class="mt-3 mb-3">
+
+## Phase Details
+
+### 1. **Startup Phase**
+
+**Purpose**: Initialize the Kubeflow Trainer Extension Framework and set up necessary components for managing training jobs.
+
+- **Internal APIs**:
+  - **Initialize Kubeflow TrainerFrameworkPipeline**: Sets up the entire Kubeflow Trainer Extension Framework.
+  - **TrainJobController**: Configures the TrainJob controller and registers it with the Manager.
+  - **Built-in Webhook Servers**: Initializes Admission Webhook Servers that handle job creation and updates.
+  - **Start Manager**: Starts the main management process.
+
+- **Extension Point**:
+  - **WatchExtension**: Registers custom reconciler builders that watch specific resources and trigger TrainJob reconciliations as necessary.
+
+### 2. **PreExecution Phase**
+
+**Purpose**: Triggered when a `TrainJob` is created or updated, this phase validates the job and prepares it for execution.
+
+- **Extension Point**:
+  - **CustomValidation**: Registers custom validation logic to validate resources before a `TrainJob` is executed. This can include checking specific fields or configurations before proceeding.
```

Comment:
```suggestion
  - **CustomValidation**: Registers custom validation logic to validate resources before a TrainJob is executed. This can include checking specific fields or configurations before proceeding.
```